browser/brave_stats/brave_stats_updater.cc

```diff
@@ -0,0 +1,375 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_stats/brave_stats_updater.h"
+
+#include <utility>
+
+#include "base/command_line.h"
+#include "base/system/sys_info.h"
+#include "brave/browser/brave_stats/brave_stats_updater_params.h"
+#include "brave/browser/brave_stats/brave_stats_updater_util.h"
+#include "brave/browser/brave_stats/switches.h"
+#include "brave/browser/version_info.h"
+#include "brave/common/brave_channel_info.h"
+#include "brave/common/network_constants.h"
+#include "brave/common/pref_names.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/net/system_network_context_manager.h"
+#include "chrome/common/channel_info.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/browser_thread.h"
+#include "net/base/load_flags.h"
+#include "net/base/url_util.h"
+#include "net/http/http_response_headers.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+
+#include "brave/components/brave_referrals/buildflags/buildflags.h"
+#if BUILDFLAG(ENABLE_BRAVE_REFERRALS)
+#include "brave/components/brave_referrals/common/pref_names.h"
+#endif
+
+// Ping the update server shortly after startup (units are seconds).
```

Comment:
pls make these constants constexpr and put them into anonymous namespace (enclosed into `brave_stats` namespace)